blob/azureblob/example_test.go

```diff
@@ -25,74 +29,215 @@ import (
+		fmt.Println("ReadAll failed due to invalid credentials")
+	}
+
+	// Output:
+	// ReadAll failed due to invalid credentials
+}
+
+func Example_open() {
+
+	// Open creates a *Bucket from a URL.
+	// Example URL: azblob://my-bucket?cred_path=replace-with-path-to-credentials-file
+	// The schema "azblob" represents the provider for Azure Blob (Azure Storage Account).
+	// The host "my-bucket" represents the blob bucket (Azure Storage Account's Container).
+	// The queryString "cred_path: represents the credentials file in JSON format. See the package documentation for credentials file schema.
+
+	_, err := blob.Open(context.Background(), "azblob://my-bucket?cred_path=replace-with-path-to-credentials-file")
+	if err != nil {
+		// Windows will return error: replace-with-path-to-credentials-file: The system cannot find the file specified.
+		// macOS and Linux will return error: open replace-with-path-to-credentials-file: no such file or directory
+		fmt.Println(err)
+	}
+	// Output:
+	// open replace-with-path-to-credentials-file: no such file or directory
+}
+
+func Example_as() {
+
+	ctx := context.Background()
+	// Set a fake key which must be base64 encoded. The Azure Storage Access Keys are already base64 encoded.
+	accountName, accountKey := "gocloudblobtests", base64.StdEncoding.EncodeToString([]byte("FAKECREDS"))
+	bucketName, key := "my-bucket", "my-key"
+
+	// Initializes an azblob.ServiceURL which represents the Azure Storage Account using Shared Key authorization.
+	serviceURL, err := azureblob.ServiceURLFromAccountKey(accountName, accountKey)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	// Credential represents the authorizer for SignedURL.
+	creds, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
 
-	// Obtain the native blob attributes
+	// Override the default retry policy so calls can return promptly for this test. Please review the timeout guidelines and set accordingly.
+	// See https://docs.microsoft.com/en-us/rest/api/storageservices/setting-timeouts-for-blob-service-operations for more information.
+	pipelineOpts := azblob.PipelineOptions{
+		Retry: azblob.RetryOptions{
+			Policy:        azblob.RetryPolicyFixed,
+			TryTimeout:    10 * time.Second,
+			MaxTries:      1,
+			RetryDelay:    0 * time.Second,
+			MaxRetryDelay: 0 * time.Second,
+		},
+	}
+	p := azblob.NewPipeline(creds, pipelineOpts)
+	*serviceURL = serviceURL.WithPipeline(p)
+
+	// Set credential for the SignedURL operation.
+	azureOpts := azureblob.Options{
+		Credential: creds,
+	}
+
+	// Create a *blob.Bucket.
+	b, err := azureblob.OpenBucket(ctx, serviceURL, bucketName, &azureOpts)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	// Create a *blob.Reader.
+	r, err := b.NewReader(ctx, key, &blob.ReaderOptions{})
+	if err != nil {
+		// This is expected due to the fake credentials we used above.
+		fmt.Println("ReadAll failed due to invalid credentials")
+		return
+	}
+
```

Comment:
Can you add a comment here clarifying that nothing below here actually runs in the example?

Reply:
Ok added.